src/atoms/Heading/Heading.web.js

```diff
@@ -1,5 +1,137 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { getColorKeys, getColor } from '../../_helpers/theme';
+import baseTheme from '../../tokens/theme';
+import automation from '../../_helpers/automation-attributes';
 
-const Heading = () => <div>WIP</div>;
+const styles = {
+  fontSize({ theme, size }) {
+    switch (size) {
+      case 'medium':
+        return theme.fonts.size.medium;
+      case 'large':
+        return theme.fonts.size.large;
+      case 'xlarge':
+        return theme.fonts.size.xxlarge;
+      case 'xxlarge':
+        return theme.fonts.size.xxxlarge;
+      case 'xxxlarge':
+        return theme.fonts.size.xxxxlarge;
+      default:
+        return theme.fonts.size.xxxxlarge;
+    }
+  },
+  lineHeight({ theme, size }) {
+    switch (size) {
+      case 'medium':
+        return theme.fonts.lineHeight.medium;
+      case 'large':
+        return theme.fonts.lineHeight.medium;
+      case 'xlarge':
+        return theme.fonts.lineHeight.xlarge;
+      case 'xxlarge':
+        return theme.fonts.lineHeight.xxlarge;
+      case 'xxxlarge':
+        return theme.fonts.lineHeight.xxlarge;
+      default:
+        return theme.fonts.lineHeight.xxlarge;
+    }
+  },
+  maxHeight({ theme, maxLines, size }) {
+    let _lineHeight;
+    switch (size) {
+      case 'medium':
+        _lineHeight = theme.fonts.lineHeight.medium;
+        break;
+      case 'large':
+        _lineHeight = theme.fonts.lineHeight.medium;
+        break;
+      case 'xlarge':
+        _lineHeight = theme.fonts.lineHeight.xlarge;
+        break;
+      case 'xxlarge':
+        _lineHeight = theme.fonts.lineHeight.xxlarge;
+        break;
+      case 'xxxlarge':
+        _lineHeight = theme.fonts.lineHeight.xxlarge;
+        break;
+      default:
+        _lineHeight = theme.fonts.lineHeight.xxlarge;
+    }
+    const _maxHeight = `${parseFloat(_lineHeight) * maxLines}px`;
+    return _maxHeight;
+  },
+};
+
+const StyledHeading = styled.div`
+  &&& {
+    font-family: ${({ theme, weight }) => theme.fonts.family.lato[weight]};
+    font-size: ${styles.fontSize};
+    font-weight: ${({ theme, weight }) => theme.fonts.weight[weight]};
+    color: ${({ theme, color }) => getColor(theme, color)};
+    letter-spacing: 0;
+    line-height: ${styles.lineHeight};
+    overflow: hidden;
+    text-overflow: ellipsis;
+    max-height: ${styles.maxHeight};
+    position: relative;
+    padding-right: 1em;
+    &&:before {
```

Comment:
Please don't do this. It should extend from Text. Even though text is broken right now it needs to be fixed. That's the whole point of a design system. The base should be fixed so all the components automatically gets fixed

Reply:
Using Text